src/Event/Event.php

```diff
@@ -172,22 +179,26 @@ public function getData(?string $key = null)
             return $this->_data[$key] ?? null;
         }
 
-        return (array)$this->_data;
+        return $this->_data;
     }
 
     /**
      * Assigns a value to the data/payload of this event.
      *
-     * @param array|string $key An array will replace all payload data, and a key will set just that array item.
+     * @param array|\ArrayAccess|string $key An array will replace all payload data, and a key will set just that array item.
      * @param mixed $value The value to set.
      * @return $this
+     * @throws \InvalidArgumentException When $key is not a string, array or does not implement \ArrayAccess.
      */
     public function setData($key, $value = null)
     {
-        if (is_array($key)) {
-            $this->_data = $key;
-        } else {
+        if (is_string($key)) {
```

Comment:
While it would be unwise one could use integer/numeric keys too.

Reply:
True, but the $key typehint was already `string` so I used that.